src/main/kotlin/org/gradle/script/lang/kotlin/provider/KotlinScriptPlugin.kt

```diff
@@ -39,6 +39,10 @@ class KotlinScriptPlugin(
         if (target !is Project) {
             throw IllegalArgumentException("target $target was not a Project as expected")
         }
+        with(target) {
+            if (!plugins.hasPlugin(KotlinScriptBasePlugin::class.java))
+                plugins.apply(KotlinScriptBasePlugin::class.java)
```

Comment:
`PluginManager#apply(Class<*>)` is idempotent so `hasPlugin` is unnecessary here.